indexer_app/utils.py

```
@@ -876,27 +1195,1017 @@
+
+async def handle_new_group(data: dict, created_at: datetime):
+    logger.info(f"new group data: {data}")
+    group_data = data.get("group", {})
+    try:
+        # group enums can have values, they are represented as a dict in the events from the indexer, and enum choices without values are presented as normal strings:
+        # withValue: {'group': {'id': 5, 'name': 'Here we go again', 'providers': [8, 1, 4, 6], 'rule': {'IncreasingReturns': 10}}}
+        # noValue: {"id":6,"name":"Lachlan test group","providers":[1,2],"rule":"Highest"}
+        rule = group_data["rule"]
+        rule_key = rule
+        rule_val = None
+        if type(rule) == dict:
```

Ruff (0.12.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)